src/event_bus.cpp

```diff
@@ -83,3 +86,308 @@ void event_bus::send_with_talker( Creature *alpha, item_location *beta,
         s->notify( e, get_talker_for( alpha ), get_talker_for( beta ) );
     }
 }
+
+void event_bus::send_dynamic( event_type type, std::vector<std::string> &args ) const
+{
+    switch( type ) {
```

Comment:
The standard trick to avoid a giant switch statement here is as follows:
- Define a static helper function which takes a compile-time integer template argument and the vector of regular arguments.  It casts that integer to the enum and calls `make_dyn` as is done in each branch of the switch here.
- Define *another* helper function which takes a `std::integer_sequence` argument and expands that into an `constexpr` array of function pointers, each of which is a pointer to an instantiation of the above function, but with a specific chosen integer value.
- `send_dynamic` can pick the appropriate function pointer from the array according to the given `type` and call it.

There's an example of this pattern in `cata_variant::is_valid`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/f2c8c58036e1a48099055586e77fe175e9f4cca4/src/cata_variant.cpp#L5-L28

Reply:
Really cool. Thank you so much!!